internal/policies/mount/mount.go

```diff
@@ -0,0 +1,215 @@
+	muMu     sync.Mutex
+
+	runDir string
+
+	userLookup   func(string) (*user.User, error)
+	systemCtlCmd []string
+}
+
+// New creates a Manager to handle mount policies.
+func New(runDir string, opts ...Option) (m *Manager, err error) {
+	defer decorate.OnError(&err, i18n.G("failed to create new mount manager"))
+	o := options{
+		userLookup: user.Lookup,
+	}
+
+	for _, opt := range opts {
+		opt(&o)
+	}
+
+	// Multiple users will be in users/ subdirectory. Create the main one.
+	// #nosec G301 - multiple users will be in users/ subdirectory, we want all of them to be able to access its own subdirectory.
+	if err := os.MkdirAll(filepath.Join(runDir, "users"), 0750); err != nil {
+		return nil, err
+	}
+
+	return &Manager{
+		mountsMu: make(map[string]*sync.Mutex),
+
+		runDir:       runDir,
+		userLookup:   o.userLookup,
+		systemCtlCmd: o.systemctlCmd,
+	}, nil
+}
+
+// ApplyPolicy generates mount policies based on a list of entries.
+func (m *Manager) ApplyPolicy(ctx context.Context, objectName string, isComputer bool, entries []entry.Entry) (err error) {
+	defer decorate.OnError(&err, i18n.G("can't apply mount policy to %s"), objectName)
+
+	if len(entries) == 0 {
+		return nil
+	}
+
+	log.Debugf(ctx, "Applying mount policy to %s", objectName)
+
+	// Mutexes are per user1, user2, computer
+	m.muMu.Lock()
+	if _, exists := m.mountsMu[objectName]; !exists {
+		m.mountsMu[objectName] = &sync.Mutex{}
+	}
+	m.muMu.Unlock()
+	m.mountsMu[objectName].Lock()
+	defer m.mountsMu[objectName].Unlock()
+
+	for _, entry := range entries {
+		switch entry.Key {
+		case "user-mounts":
+			if isComputer {
+				break
+			}
+
+			if e := m.applyUserPolicy(ctx, objectName, entry); e != nil {
+				err = e
+			}
+
+		default:
+			log.Debugf(ctx, "Key %q is currently not supported by the mount manager", entry.Key)
+		}
+	}
+
+	return err
+}
+
+func (m *Manager) applyUserPolicy(ctx context.Context, username string, entry entry.Entry) (err error) {
+	defer decorate.OnError(&err, i18n.G("failed to apply policy for user %q"), username)
+
+	log.Debugf(ctx, "Applying mount policy to user %q", username)
+
+	var uid, gid int
+	usr, err := m.userLookup(username)
+	if err != nil {
+		return fmt.Errorf(i18n.G("could not retrieve user for %q: %w"), err)
+	}
+	if uid, err = strconv.Atoi(usr.Uid); err != nil {
+		return fmt.Errorf(i18n.G("couldn't convert %q to a valid uid for %q"), usr.Uid, username)
+	}
+	if gid, err = strconv.Atoi(usr.Gid); err != nil {
+		return fmt.Errorf(i18n.G("couldn't convert %q to a valid gid for %q"), usr.Gid, username)
+	}
+
+	objectPath := filepath.Join(m.runDir, "users", usr.Uid)
+	mountsPath := filepath.Join(objectPath, "mounts")
+
+	// Removes the current mounts file, if it exists, before continuing applying the policy.
+	if err = os.Remove(mountsPath); err != nil && !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+
+	// This creates userDir directory.
+	// We chown userDir to uid:gid of the user. Nothing is done for the machine
```

Comment:
As there is no more userDir variable, you can write:
`This creates user directory and chown it to uid:gid of the user`

Also, there is no need anymore for the "Nothing is done for the machine", now that you split it in a deddicated user function.